lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -77,6 +77,7 @@ internal sealed class IMRUDriver<TMapInput, TMapOutput, TResult> : IObserver<IDr
         private int _currentFailedEvaluators = 0;
         private bool _reachedUpdateTaskActiveContext = false;
         private readonly bool _invokeGC;
+        private readonly IList<IActiveContext> _activeContexts = new List<IActiveContext>();
```

Comment:
stack may not work because we will need to allow any element to be removed if evaluator/context fail. It can be set, list, definitely need to be locked. 
We can use a thread safe collection, or just  lock for now and see if we need a common one when we move on. 
I will add lock for its operations.